### PR TITLE
test(operator): Add comprehensive unit tests for trustee module

### PR DESCRIPTION
**Pull request by 6-dehan**
This commit introduces a full suite of 15 unit tests for the functions within `trustee.rs`, validating pure logic, error handling, and both simple and complex Kubernetes API interactions using a mocked client.

-   `test_get_image_pcrs_success`: Verifies that a valid JSON string in a ConfigMap is correctly deserialized into the `ImagePcrs` struct.
-   `test_get_image_pcrs_no_data`: Ensures an error is returned when the ConfigMap's `data` field is missing.
-   `test_get_image_pcrs_invalid_json`: Confirms that an error is propagated when the data contains an invalid JSON string.
-   `test_generate_luks_key_returns_correct_size`: A sanity check to validate that `generate_luks_key` runs without errors and returns a key of the expected 32-byte length.

These tests validate the idempotency and error handling of functions that perform a single `create` operation, primarily testing the `info_if_exists!` macro.

-   `test_create_rv_config_map_success`: Verifies the function returns `Ok(())` on a successful API response (200 OK).
-   `test_create_rv_config_map_already_exists`: Verifies the function correctly handles a 409 Conflict and returns `Ok(())`, confirming idempotency.
-   `test_create_rv_config_map_generic_error`: Ensures a generic API error (e.g., 500) is properly propagated as an `Err`.
-   `test_generate_resource_policy_success`: Validates the success path for the `generate_resource_policy` function.
-   `test_generate_kbs_https_certificate_success`: Validates the success path for the `generate_kbs_https_certificate` function.
-   `test_generate_kbs_configurations_success`: Validates the success path for the `generate_kbs_configurations` function.
-   `test_generate_attestation_policy_success`: Validates the success path for the `generate_attestation_policy` function.
-   `test_generate_kbs_success`: Validates the success path for the `generate_kbs` function.

These tests use a stateful mock client to simulate entire operational flows involving multiple API calls.

-   `test_recompute_reference_values_flow`: Verifies the complete `GET (PCRs) -> GET (RV map) -> PUT (RV map)` sequence executes successfully.
-   `test_generate_secret_flow_success`: Validates the full `CREATE (Secret) -> GET (KbsConfig) -> PATCH (KbsConfig)` workflow for adding a new secret.
-   `test_generate_secret_already_present_in_spec`: Tests the boundary condition where a secret ID already exists in the KbsConfig spec, ensuring the function exits early without making a redundant PATCH call.

### CODE REVIEW

File: operator/src/trustee.rs [hunk 1]
```diff
@@ -151,21 +153,30 @@ pub async fn generate_kbs_configurations(
...
-    let config_map = ConfigMap {
-        metadata: kube::api::ObjectMeta {
-            name: Some(trustee.kbs_configuration.to_string()),
-            namespace: Some(namespace.to_string()),
+    let as_config = include_str!("as-config.json");
+    let rvps_config = include_str!("rvps-config.json");
+
+    for (filename, content, configmap) in [
+        ("kbs-config.toml", kbs_config, &trustee.kbs_configuration),
+        ("as-config.json", as_config, &trustee.as_configuration),
```
Comment (Jakob-Naucke) [inline operator/src/trustee.rs:161]: This PR version has some of the merged changes restored, but not all of them (e.g. this older piece is still there). I don't think this PR needs to change most of what's above the `tests` module, just the `Deserialize` derive (I could be missing something that is required though).

File: operator/src/trustee.rs [hunk 2]
```diff
@@ -176,10 +187,12 @@ pub async fn recompute_reference_values(ctx: RvContextData) -> anyhow::Result<()
     let image_pcrs_map = operator_config_maps.get(PCR_CONFIG_MAP).await?;
     let image_pcrs = get_image_pcrs(image_pcrs_map)?;
     // TODO many grub+shim:many OS image recompute once supported
-    let mut reference_values_in = BTreeMap::from([(
-        "svn".to_string(),
-        vec![serde_json::Value::String("1".to_string())],
-    )]);
+    let mut reference_values_in = BTreeMap::from([
+        (
```
Comment (Jakob-Naucke) [inline operator/src/trustee.rs:191]: There are a lot of changes in the PR that are formatting only (e.g. this one), but the code was formatted right, which is why the format CI is failing. Please use `cargo fmt` with a recent version (1.85+) and default config.

**Review by Jakob-Naucke — COMMENTED** [review]
Thanks for your updates. Please ensure that the CI passes, e.g. formatting your code using `cargo fmt`.
